src/main/scala/beam/agentsim/infrastructure/ChargingFunctions.scala

```diff
@@ -49,11 +50,11 @@ class ChargingFunctions(
     * @param inquiry ParkingInquiry
     * @return
     */
-  def ifRideHailCurrentlyOnShiftThenFastChargingOnly(zone: ParkingZone, inquiry: ParkingInquiry): Boolean = {
+  private def ifRideHailCurrentlyOnShiftThenFastChargingOnly(zone: ParkingZone, inquiry: ParkingInquiry): Boolean = {
     zone.chargingPointType.forall(chargingPointType =>
       if (
-        inquiry.reservedFor.managerType == VehicleManager.TypeEnum.RideHail || inquiry.beamVehicle
-          .exists(v => v.isRideHail && (inquiry.parkingDuration <= 3600 || v.isCAV))
+        inquiry.parkingDuration <= 3600 && (inquiry.reservedFor.managerType == VehicleManager.TypeEnum.RideHail || inquiry.beamVehicle
+          .exists(_.isRideHail))
```

Comment:
This lost the CAV check - was that on purpose?

Reply:
I have updated the whole section